lib/tree.rb

```diff
@@ -2,59 +2,133 @@ class TreeNode
   attr_reader :key, :value
   attr_accessor :left, :right
 
-   def initialize(key, val)
+  def initialize(key, val)
     @key = key
     @value = val
     @left = nil
     @right = nil
-   end
+  end
 end
 
 class Tree
   attr_reader :root
+
   def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(n)
+  # Space Complexity: O(n)
   def add(key, value)
```

Comment:
👍 , Time complexity is O(log n) if the tree is balanced and O(n) if it's unbalanced.  Same for space complexity due to the system stack.